### PR TITLE
Add a new section on themes empty results to upgrade the plan

### LINKED ISSUE #68285: Marketplace Search: Add an “Astra” theme card that will have a CTA “Install by upgrading to Business plan”
### Details 

Add an [“Astra” theme card](https://wordpress.org/themes/astra/) that will have a CTA “Install by upgrading to Business plan”. Clicking this CTA will lead the user through checkout and then theme installation as we already do with plugins.

Related to pdgAPm-uo-p2#comment-446

Design: @SaxonF suggested (pdgAPm-uo-p2#comment-453) showing a more generic card for all empty searches like the following 

image

Let's follow this approach for `Astra` terms and follow up on how we can expand it to more themes - the challenge is that the theme searched should exist in the [WP.org directory](https://wordpress.org/themes).

### PR DESCRIPTION
**Pull request by WBerredo**
#### Proposed Changes

* Add a section to upsell the Business plan when no result is found
* Add a button that redirects to the cart with the business plan
* Add a track event for the upgrade plan button

Screen Shot 2022-10-06 at 18 08 08
 

#### Testing Instructions
* Go to the themes page on a site with a free plan  (`/themes/{free_site}`)
* Search for a term that shouldn't return any theme. Ex: `not found`
* Check if the section with an upgrade button is shown.

---

* Go to the themes page on a site with a paid plan(`/themes/{paid_sote}`)
* Search for a term that shouldn't return any theme. Ex: `not found`
* Check if the section with an upgrade button is similar to what is currently in production

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Fixes #68285

### CODE REVIEW

File: client/components/themes-list/style.scss [hunk 1]
```diff
@@ -17,3 +17,39 @@
...
+		box-shadow: 0 0 0 1px var(--color-border-subtle);
+	}
+
+	.themes-list__upgrade-section-title {
+		font-size: $font-title-small;
+		color: var(--studio-gray-100);
+	}
+
+	.themes-list__upgrade-section-subtitle {
+		width: 680px;
```
Comment (epeicher) [inline client/components/themes-list/style.scss]: This fixes the small overflow when resizing the screen and provides a bit of padding to the text for small screens
```suggestion
		max-width: 680px;
	   padding: 0 2rem;
```

File: client/components/themes-list/style.scss [hunk 2]
```diff
@@ -17,3 +17,39 @@
 	margin: 0 10px;
 	flex-grow: 1;
 }
+
+.themes-list__empty-container {
+	.themes-list__not-found-text {
+		color: var(--studio-gray-60);
+		margin-top: 40px;
+		margin-bottom: 20px;
```
Comment (epeicher) [inline client/components/themes-list/style.scss]: Those margins are slightly different to the Figma design although the variance is really small so I will leave it to you to decide 😄 
CleanShot 2022-10-07 at 16 28 34@2x
CleanShot 2022-10-07 at 16 28 23@2x

**Review by epeicher — COMMENTED** [review]
Thanks for adding the card, it looks great. 

One small issue I've found is that there is a bit of overflow for small screens with the text. I have suggested a potential fix in the comments.

One question about the nudge, should it be removed when showing the card as per Saxon comment here p1665087375607999/1664977962.532239-slack-C029JEQRVRT ? That could also be covered as a follow-up task to iterate on this one if you prefer.

CleanShot 2022-10-07 at 16 02 26@2x

File: client/components/themes-list/style.scss (hunk 1 above)
Comment (WBerredo) [reply client/components/themes-list/style.scss]: Fixed at https://github.com/Automattic/wp-calypso/pull/68542/commits/cd3857d37651e679d2aa4e45b8a615ed5fc26835

File: client/components/themes-list/style.scss (hunk 2 above)
Comment (WBerredo) [reply client/components/themes-list/style.scss]: Fixed at https://github.com/Automattic/wp-calypso/pull/68542/commits/cd3857d37651e679d2aa4e45b8a615ed5fc26835

**Comment by WBerredo** [comment]
Thanks for the review @epeicher , I applied your suggestions and I created an issue to track this nudge removal #68789.

**Comment by epeicher** [comment]
The track event has been updated to include the `site_plan` so we can group the events depending on it

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7573570

Thank you @WBerredo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7573570).

**Pull request merged**